drivers/sensor/voltage_divider/voltage_divider.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2022 Aurelien Jarno
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(voltage_divider, CONFIG_SENSOR_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(voltage_divider)
+#define DT_DRV_COMPAT voltage_divider
+#else
+#error "No compatible devicetree node found"
+#endif
+
+struct voltage_divider_data {
+	const struct adc_dt_spec adc_channel;
+	const struct gpio_dt_spec power_gpios;
+	struct adc_sequence adc_seq;
+	struct k_mutex mutex;
+	int16_t sample_buffer;
+	int16_t raw; /* raw adc Sensor value */
+};
+
+struct voltage_divider_config {
+	/* output_ohm is used as a flag value: if it is nonzero then
+	 * the voltage is measured through a voltage divider;
+	 * otherwise it is assumed to be directly connected.
+	 */
+	uint32_t output_ohms;
+	uint32_t full_ohms;
+};
+
+static int voltage_divider_enable(const struct device *dev, bool enable)
+{
```

Comment:
what is the difference when enable is true/false -> should 'enable' or disable but there is no check on 'enable' boolean